src/containers/Tenant/Diagnostics/TopicData/TopicMessageDetails/components/TopicMessage.tsx

```diff
@@ -0,0 +1,132 @@
+            convertedMessage = unipikaConvert(preparedMessage);
+        } else if (preparedMessage && typeof preparedMessage === 'object') {
+            preparedMessage = JSON.stringify(preparedMessage, null, 2);
+        }
+
+        return {preparedMessage, decodedMessage, convertedMessage};
+    }, [message, size]);
+
+    const isJson = Boolean(convertedMessage);
+
+    const messageContent = isJson ? (
+        <JsonViewer
+            // key is used to reset JsonViewer state to collapsed due to performance issues on close fullscreen mode if nodes quantity is big enough https://github.com/ydb-platform/ydb-embedded-ui/issues/2265
+            key={String(isFullscreen)}
+            collapsedInitially
+            value={convertedMessage}
+            maxValueWidth={50}
+            toolbarClassName={b('json-viewer-toolbar')}
+        />
+    ) : (
+        <div className={b('string-message')}>
+            {/* key is used to reset string's state when toggle fullscreen: otherwise if very long string is expanded, it may be performance issues on open fullscreen mode https://github.com/ydb-platform/ydb-embedded-ui/issues/2265  */}
+            <ShortyString key={String(isFullscreen)} value={preparedMessage} limit={2000} />
+        </div>
+    );
+
+    const renderToolbar = () => {
+        return (
+            <Flex gap={1}>
+                <ActionTooltip title={i18n('label_download')}>
+                    <Button
+                        view="flat-secondary"
+                        onClick={(e) => {
+                            e.stopPropagation();
+                            createAndDownloadFile(
+                                decodedMessage,
+                                `topic-message-${offset ?? 'unknown-offset'}`,
+                            );
+                        }}
+                    >
+                        <Icon data={ArrowDownToLine} />
+                    </Button>
+                </ActionTooltip>
+                <ClipboardButton view="flat-secondary" text={decodedMessage} />
+            </Flex>
+        );
+    };
+
+    const truncated = safeParseNumber(size) > MESSAGE_SIZE_LIMIT;
+
+    return (
+        <TopicDataSection
+            title={<MessageTitle truncated={truncated} />}
+            renderToolbar={renderToolbar}
+            className={b('message')}
+        >
+            {messageContent}
+        </TopicDataSection>
+    );
+}
+
+interface MessageTitleProps {
+    truncated?: boolean;
+}
+
+function MessageTitle({truncated}: MessageTitleProps) {
+    return (
+        <span>
+            {i18n('label_message')}
+            {truncated && (
+                <React.Fragment>
+                    {' '}
+                    <Text color="secondary">
+                        [
+                        {i18n('label_truncated', {
+                            size: bytesToMB(MESSAGE_SIZE_LIMIT),
+                        })}
+                        ]
```

Comment:
Redundant `[]`

Reply:
It's not redundant
<img width="881" alt="Screenshot 2025-05-16 at 16 06 52" src="https://github.com/user-attachments/assets/627b4f01-315c-4760-be92-8695343f7a5e" />